### ISSUE TITLE
No music on Oneplus5T/LineageOS

### ISSUE THREAD

**Issue by katzenpapst** [00-Evan/shattered-pixel-dungeon#340]
Got a Oneplus5T with LineageOS as a new phone, and noticed that Shattered Pixel Dungeon doesn't  play any music, despite it being unmuted.

I wonder if it's me who messed up somehow, if I found a really obscure bug, or if it's a know issue.
I restored the app from backup using TitaniumBackup, I guess something weird might have happened there. It confuses me even further, that I seemingly can uninstall the game, reinstall it from F-Droid, and my save is still there. It does get deleted if I delete the data using Android's "delete data" feature.

This doesn't happen with the "normal" pixel dungeon.

The device is rooted, and I have access to ADB, so I could definitely provide additional infos/try additional things. However, I'm not that experienced with actually debugging android apps...

**Comment by theRebelRevenant** [comment]
I have the same issue on a Oneplus 3/LineageOS 14 (rooted).

On September 8, 2019 3:43:53 PM UTC, katzenpapst <notifications@github.com> wrote:
>Got a Oneplus5T with LineageOS as a new phone, and noticed that
>Shattered Pixel Dungeon doesn't  play any music, despite it being
>unmuted.
>
>I wonder if it's me who messed up somehow, if I found a really obscure
>bug, or if it's a know issue.
>I restored the app from backup using TitaniumBackup, I guess something
>weird might have happened there. It confuses me even further, that I
>seemingly can uninstall the game, reinstall it from F-Droid, and my
>save is still there. It does get deleted if I delete the data using
>Android's "delete data" feature.
>
>This doesn't happen with the "normal" pixel dungeon.
>
>The device is rooted, and I have access to ADB, so I could definitely
>provide additional infos/try additional things. However, I'm not that
>experienced with actually debugging android apps...
>
>
>-- 
>You are receiving this because you are subscribed to this thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/00-Evan/shattered-pixel-dungeon/issues/340

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by 00-Evan** [comment]
This sounds like it's likely an issue with LineageOS, and I have occasionally heard about music issues from users on custom ROMS. The best info you could provide would be some logcat output. Try going into the game and muting/unmuting the music and couple times and paste any resulting logcat output.

**Comment by katzenpapst** [comment]
Uh. I tried. There's a lot of spam happening as soon as I unmute.
https://pastebin.com/5h6e1zm8

I see this:
```
msm8974_platform: platform_check_backends_match: Invalid snd_device = 
09-08 22:59:21.293   779  1363 E qcbassboost: 	[vendor/qcom/proprietary/mm-audio-noship/audio-effects/safx/android-adapter/qcbassboost/qcbassboost.c:536] Assertion fail: pConfig->inputCfg.format == AUDIO_FORMAT_PCM_16_BIT
```
Now the latter I got rid of by disabling some "audio effects" app. But it still doesn't work.

**Comment by 00-Evan** [comment]
yeah there will be log spam. You're only sending 0.6 seconds of logcat info here though, which may not be enough. Can you try again with these steps:

- turn music on ingame
- force-stop Shattered Pixel Dungeon
- start recording logcat
- open Shattered
- stop recording logcat

And then send the entire logcat output, even if it's huge.

**Comment by katzenpapst** [comment]
Okay, next attempt:
https://pastebin.com/xdm9xutn

didn't run it for long, though, pressed ctrl+c in the console spam almost right away.

**Comment by 00-Evan** [comment]
Hmm, from taking an initial skim of that log I don't see anything super unusual.

What version of LineageOS are you running, and would you be willing to try some test APKs for me to see if we can isolate the issue?

**Comment by katzenpapst** [comment]
Lineage is 16.0. And yes, I can run some test APKs.

I also reported this on the Lineage gitlab (at least I hope I did it properly...)

**Comment by 00-Evan** [comment]
Sorry for the delay in getting back to you on this, been busy working on the next update.

Here is an APK which is identical to the current open-source (and therefore to the F-droid version), except I have reverted the music loading/playing functionality back to the system used in v0.7.4. Please let me know if you're able to get music on this version. If you are, we can start working forwards and figure out which specific change caused the problem.

[ShatteredPD v0.7.4d - MUSIC_DEBUG_1.zip](https://github.com/00-Evan/shattered-pixel-dungeon/files/3632588/ShatteredPD.v0.7.4d.-.MUSIC_DEBUG_1.zip)

Note that this is signed with my dev key, so you will need to uninstall Shattered first if you're using the f-droid version.

**Comment by theRebelRevenant** [comment]
I just upgraded to the latest daily of lineage 16 on a Oneplus 3, and the zip you provided still didn't have music. There was a time though, with the fdroid apk, where music suddenly played. I had started playing some music from vlc for bg music, when the shattered music began. I turned off vlc, and the shattered music remained until I had turned off my phone and rebooted.

On September 19, 2019 10:02:17 AM AKDT, Evan Debenham <notifications@github.com> wrote:
>Sorry for the delay in getting back to you on this, been busy working
>on the next update.
>
>Here is an APK which is identical to the current open-source (and
>therefore to the F-droid version), except I have reverted the music
>loading/playing functionality back to the system used in v0.7.4. Please
>let me know if you're able to get music on this version. If you are, we
>can start working forwards and figure out which specific change caused
>the problem.
>
>[ShatteredPD v0.7.4d -
>MUSIC_DEBUG_1.zip](https://github.com/00-Evan/shattered-pixel-dungeon/files/3632588/ShatteredPD.v0.7.4d.-.MUSIC_DEBUG_1.zip)
>
>Note that this is signed with my dev key, so you will need to uninstall
>Shattered first if you're using the f-droid version.
>
>-- 
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/00-Evan/shattered-pixel-dungeon/issues/340#issuecomment-533243552

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by 00-Evan** [comment]
Hmm, interesting. I also made changes to the 'call listener' functionality in 0.7.4b, which attempts to mute game music when a device phone call is active. That might also explain the odd behavior. Here's another APK with all 0.7.4d code except that call state listener is disabled:

[ShatteredPD v0.7.4d - MUSIC_DEBUG_2.zip](https://github.com/00-Evan/shattered-pixel-dungeon/files/3632794/ShatteredPD.v0.7.4d.-.MUSIC_DEBUG_2.zip)

**Comment by theRebelRevenant** [comment]
this one doesn't work either. I tried to recreate the instance where vlc caused it to start and couldn't. I've also been restarting before I attempt the new apk.

On September 19, 2019 11:04:58 AM AKDT, Evan Debenham <notifications@github.com> wrote:
>Hmm, interesting. I also made changes to the 'call listener'
>functionality in 0.7.4b, which attempts to mute game music when a
>device phone call is active. That might also explain the odd behavior.
>Here's another APK with all 0.7.4d code except that call state listener
>is disabled:
>
>[ShatteredPD v0.7.4d -
>MUSIC_DEBUG_2.zip](https://github.com/00-Evan/shattered-pixel-dungeon/files/3632794/ShatteredPD.v0.7.4d.-.MUSIC_DEBUG_2.zip)
>
>
>-- 
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/00-Evan/shattered-pixel-dungeon/issues/340#issuecomment-533266544

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by katzenpapst** [comment]
Neither of those two work for me either.

I upgraded to today's Lineage update, too.

**Comment by 00-Evan** [comment]
Hmm, alright, then let's try this. Shattered currently uses .ogg files for its music, as they are more space efficient than MP3. Perhaps there's some issues with .ogg playback on lineageOS.

Here is another APK, identical to 0.7.4d, except that the music files are replaced with the .MP3 ones from vanilla:
[ShatteredPD v0.7.4d - MUSIC_DEBUG_3.zip](https://github.com/00-Evan/shattered-pixel-dungeon/files/3632926/ShatteredPD.v0.7.4d.-.MUSIC_DEBUG_3.zip)

**Comment by theRebelRevenant** [comment]
ok yea, that did it for me. too bad, I like ogg format more in general :(

On September 19, 2019 11:46:47 AM AKDT, Evan Debenham <notifications@github.com> wrote:
>Hmm, alright, then let's try this. Shattered currently uses .ogg files
>for its music, as they are more space efficient than MP3. Perhaps
>there's some issues with .ogg playback on lineageOS.
>
>Here is another APK, identical to 0.7.4d, except that the music files
>are replaced with the .MP3 ones from vanilla:
>[ShatteredPD v0.7.4d -
>MUSIC_DEBUG_3.zip](https://github.com/00-Evan/shattered-pixel-dungeon/files/3632926/ShatteredPD.v0.7.4d.-.MUSIC_DEBUG_3.zip)
>
>
>-- 
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/00-Evan/shattered-pixel-dungeon/issues/340#issuecomment-533281099

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by katzenpapst** [comment]
Same here. That works.

Guess OGG playback really is broken in LineageOS :(

**Comment by 00-Evan** [comment]
Well that's some progress at least! It sounds like this is an issue you should raise with the LineageOS team, feel free to reference this issue when you do so.

Unfortunately I don't currently plan to remove the existing .ogg files, though I may do so in the future as .ogg playback is not supported on iOS.

**Comment by katzenpapst** [comment]
I just found a viable workaround based on this:
https://review.lineageos.org/c/LineageOS/android_device_sony_nile-common/+/244261

tl;dr /vendor/etc/audio_policy_configuration.xml, remove or comment out the part about AUDIO_FORMAT_VORBIS. 

I made a Magisk module which seems to work for me, anyone who wants to test it message me please. I'm not sure if I should attempt to put it into the Magisk repo yet...

**Comment by foundObjects** [comment]
> I just found a viable workaround based on this:
> https://review.lineageos.org/c/LineageOS/android_device_sony_nile-common/+/244261
> 
> tl;dr /vendor/etc/audio_policy_configuration.xml, remove or comment out the part about AUDIO_FORMAT_VORBIS.
> 
> I made a Magisk module which seems to work for me, anyone who wants to test it message me please. I'm not sure if I should attempt to put it into the Magisk repo yet...

I wouldn't mind taking a look at the module. I'm also having ogg issues on lineage 16.

**Comment by katzenpapst** [comment]
OK, I just uploaded the module: https://github.com/katzenpapst/lineage-ogg-fix
Absolute noob's work, though
